### ISSUE TITLE
Notice when shellcheck disable does not apply

### ISSUE THREAD

**Issue by stdedos** [koalaman/shellcheck#1863]
#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

# shellcheck disable=SC2016
grep -oP '(?<=export PS1="\[)\d+(?=\] \{PS1}")' "${HOME}/.bashrc" > "${HOME}/.id"
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

> Note: SC2016 is not detected in this line/file/block/function/context. Consider removing it.

It could be an opt-in feature, if people disagree with the pedantic cleanup.

**Comment by matthewpersico** [comment]
I’m sure this is a duplicate.

**Comment by matthewpersico** [comment]
See #1719

**Comment by stdedos** [comment]
Good one. For some reason, "directive" did not cross my mind (and I don't know how the owner would feel about _warn_)
